pilot/pkg/bootstrap/server.go

```diff
@@ -327,6 +327,7 @@ func (s *Server) getKubeCfgFile(args *PilotArgs) (kubeCfgFile string) {
 // initKubeClient creates the k8s client if running in an k8s environment.
 func (s *Server) initKubeClient(args *PilotArgs) error {
 	needToCreateClient := false
+	// FIXME: in the case of multiple registries, we are currently only checking needToCreateClient for the last one
```

Comment:
add a break anyplace you set `needToCreateClient = true`?

Reply:
break the for loop with a label, otherwise it just breaks the switch statement.